### PR TITLE
Display a warning when the IG account is not connected to a Facebook page

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Related to 1203820938137329-as-1204561445874880/f

## Proposed Changes

* Display a warning when the IG account is not connected to a Facebook page

## Testing Instructions

* Ensure that you have D111103-code applied to your sandbox
* Add the blog sticker to your site as mentioned in D111103-code
* Point `public-api.wordpress.com` to your sandbox
* Goto `/marketing/connections/:site`
* Try to connect an Instagram account that is NOT linked to a Facebook page
* Confirm that you see the warning `'No Instagram business accounts linked to your Facebook pages found.'`
* Now try to connect with an Instagram account that IS connected to a Facebook page
* Confirm that you do not see the warning
* Try to disconnect the account
* Confirm that you do not see the warning

Screenshot 2023-06-01 at 10 35 45 AM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/marketing/connections/services/instagram-business.jsx [hunk 1]
```diff
@@ -17,6 +17,20 @@ export class InstagramBusiness extends SharingService {
 		/* eslint-disable wpcalypso/jsx-classname-namespace */
 		<SocialLogo icon="instagram" size={ 48 } className="sharing-service__logo" />
 	);
+
+	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
+	UNSAFE_componentWillReceiveProps( { availableExternalAccounts } ) {
+		if ( this.state.isAwaitingConnections && ! availableExternalAccounts.length ) {
+			this.props.errorNotice(
+				this.props.translate(
+					'Instagram business account could not be connected because we could not find any Instagram business account linked to a Facebook page.'
```
Comment (pablinos) [inline client/my-sites/marketing/connections/services/instagram-business.jsx]: This seems quite long, but I'm struggling to make something more succinct. Maybe "No Instagram business accounts linked to your Facebook pages found."?

File: client/my-sites/marketing/connections/services/instagram-business.jsx [hunk 2]
```diff
@@ -17,6 +17,20 @@ export class InstagramBusiness extends SharingService {
 		/* eslint-disable wpcalypso/jsx-classname-namespace */
 		<SocialLogo icon="instagram" size={ 48 } className="sharing-service__logo" />
 	);
+
+	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
+	UNSAFE_componentWillReceiveProps( { availableExternalAccounts } ) {
```
Comment (pablinos) [inline client/my-sites/marketing/connections/services/instagram-business.jsx:22]: Seems a shame that we need to add another instance of this, but we'd have to refactor SharingService to get rid of this.

**Review by pablinos — COMMENTED** [review]
This works well for me, I just wondered if we should make the message a little shorter.

File: client/my-sites/marketing/connections/services/instagram-business.jsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/my-sites/marketing/connections/services/instagram-business.jsx]: > Maybe "No Instagram business accounts linked to your Facebook pages found."?

Sounds good.

File: client/my-sites/marketing/connections/services/instagram-business.jsx (hunk 2 above)
Comment (manzoorwanijk) [reply client/my-sites/marketing/connections/services/instagram-business.jsx:22]: Right.

File: client/my-sites/marketing/connections/services/instagram-business.jsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/my-sites/marketing/connections/services/instagram-business.jsx]: Done. Thanks.

**Review by pablinos — APPROVED** [review]
Tests are failing, but I think this is good to merge.

**Comment by manzoorwanijk** [comment]
> Tests are failing

I re-ran the tests and all is good now ✅

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8019852

Thank you @manzoorwanijk for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8019852).

**Pull request merged**